utils/logging/log.go

```diff
@@ -294,73 +296,91 @@ func (l *Log) SetLogLevel(lvl Level) {
 	l.config.LogLevel = lvl
 }
 
-// SetDisplayLevel ...
+// SetDisplayLevel implements the Logger interface
 func (l *Log) SetDisplayLevel(lvl Level) {
 	l.configLock.Lock()
 	defer l.configLock.Unlock()
 
 	l.config.DisplayLevel = lvl
 }
 
-// SetPrefix ...
+// SetPrefix implements the Logger interface
 func (l *Log) SetPrefix(prefix string) {
 	l.configLock.Lock()
 	defer l.configLock.Unlock()
 
 	l.config.MsgPrefix = prefix
 }
 
-// SetLoggingEnabled ...
+// SetLoggingEnabled implements the Logger interface
 func (l *Log) SetLoggingEnabled(enabled bool) {
 	l.configLock.Lock()
 	defer l.configLock.Unlock()
 
 	l.config.DisableLogging = !enabled
 }
 
-// SetDisplayingEnabled ...
+// SetDisplayingEnabled implements the Logger interface
 func (l *Log) SetDisplayingEnabled(enabled bool) {
 	l.configLock.Lock()
 	defer l.configLock.Unlock()
 
 	l.config.DisableDisplaying = !enabled
 }
 
-// SetContextualDisplayingEnabled ...
+// SetContextualDisplayingEnabled implements the Logger interface
 func (l *Log) SetContextualDisplayingEnabled(enabled bool) {
 	l.configLock.Lock()
 	defer l.configLock.Unlock()
 
 	l.config.DisableContextualDisplaying = !enabled
 }
 
+// fileWriter implements the RotatingWriter interface
 type fileWriter struct {
 	writer *bufio.Writer
 	file   *os.File
 
-	config    Config
-	fileIndex int
+	config Config
 }
 
+// Flush implements the RotatingWriter interface
 func (fw *fileWriter) Flush() error {
 	return fw.writer.Flush()
 }
 
+// Write implements the RotatingWriter interface
 func (fw *fileWriter) Write(b []byte) (int, error) {
 	return fw.writer.Write(b)
 }
 
+// WriteString implements the RotatingWriter interface
 func (fw *fileWriter) WriteString(s string) (int, error) {
 	return fw.writer.WriteString(s)
 }
 
+// Close implements the RotatingWriter interface
 func (fw *fileWriter) Close() error {
 	return fw.file.Close()
 }
 
+// Rotate implements the RotatingWriter interface
 func (fw *fileWriter) Rotate() error {
-	fw.fileIndex = (fw.fileIndex + 1) % fw.config.RotationSize
-	writer, file, err := fw.create(fw.fileIndex)
+	for i := fw.config.RotationSize - 1; i >= 0; i-- {
+		sourceFilename := filepath.Join(fw.config.Directory, fmt.Sprintf("%s.log.%d", fw.config.LoggerName, i))
```

Comment:
This will create file names of structure "C.log.0". "C.log.1", etc. I think the idea was to do "C.0.log", "C.1.log" etc. Any specific reason for proposing this structure?

Reply:
Yes there were a couple of reasons,
1. I think the `filename.log.[rotation_index]` naming convention is the most common one. This is what logrotate uses by default.
2. As you mention, the original suggestion in #699 was to have log files as follows:

> |- main.0.log
> |- C.0.log
> |- X.0.log
> |- P.0.log
> |- http.0.log

However this was a problem for the chains' http logs, which used to be kept separate in a different subfolder for each chain (`C/http/0.log`, `X/http/0.log`...). I assumed this was an oversight and that the intention was not to have all http logs indifferently bundled in a single file. 
Consequently I opted to store the chains' http logs are as `[chain_name].http.log`.  
In this case, if we put the rotation index number before the `.log` extension, this potentially creates confusion: in `C.0.log`, "0" is a rotation number but could also be interpreted as a sublogger name like in `C.http.log`.

However, happy to revert to the original scheme + http logs all stored in the same file if this preferred.